# 인사

안녕하세요 ~~ 25기 여러분 반갑습니다 ~~  

드디어 정규세션이 시작되었네요.  

뭘 과제로 내볼까 고민하다가, 요즘 데이터 분석 채용 공고 자격 요건에 필수로 들어있는.... A/B TEST가 좋겠다!!고 생각했습니다.  

데이터 분석쪽이 아니더라도, 다양한 도메인에서 쓸 수 있는 방법이라고 생각해요.


A/B TEST의 개념부터 기초적인 과정을 이해해볼 수 있게 내봤어요. 근데 설명해야할 게 많다보니 텍스트가 너무 많아졌네요.

읽기 힘들어도 ㅎㅎ... 열심히 해주시고, 질문 사항은 저에게 연락 남겨주세요.  

참고로, 정규세션 발표 외의 내용을 구글링하고 공부해야만 풀 수 있는 문제들이 포함되어 있습니다.

제 이름... 아시죠...? 24기 박현서입니다....



<br>

# Q1. A/B TEST란 무엇일까요?   
또한, A/B TEST시 유의할 사항들을 초두효과를 포함해 기술해주세요.

정규세션에서 발표한 내용 외에 추가적으로 공부한 개념을 작성하셔도 됩니다.
정의와 함께 예시 하나만 들어주세요 ~~

## A1. 답안 작성  
답변: A/B 테스트는 임의화 대조군 실험의 일종으로, 주로 웹사이트나 앱 서비스에서 사용자 경험이나 비즈니스 성과를 최적화하기 위해 사용되는 데이터 분석 방법론이다.

전체 사용자를 무작위로 두 개(또는 그 이상)의 집단으로 나누고, 대조군에게는 기존 서비스를, 실험군에게는 새로운 기능을 노출한 뒤, 어떤 집단이 더 높은 성과(클릭률, 구매 전환율 등)를 보이는지 통계적으로 검증하는 과정이다.

예를 들어, 핀테크 앱 메인 화면에서 '송금하기' 버튼의 클릭률(CTR)을 높이고자 할때, 송금 버튼을 하단 탭바 중앙에 크게 배치하면, 기존 상단 배치보다 접근성이 좋아져 송금 시도 횟수가 증가할 것이라고 가정하고 대조군과 실험군에게 일일 활성 사용자 대비 송금 버튼 클릭 비율을 일정 기간 테스트한다. 결과적으로 실험군의 클릭률이 대조군보다 통계적으로 유의미하게 높다면 변경된 앱을 선택한다.

A/B 테스트 진행 시 유의할 사항으로는,

첫 번째로 초두 효과가 있다.
초두 효과는 실험 초기 단계에서 발생하는 편향을 의미하며, 신기효과와 변화 저항으로 나타날 수 있다.
신기효과는 사용자가 변경된 UI가 더 좋아서가 아니라, 단지 새롭기 때문에 호기심에 클릭해보는 현상이다. 이로 인해 초기에 성과가 일시적으로 급등했다가 시간이 지나면 원래대로 돌아간다.
반대로, 변화 저항은 익숙한 인터페이스가 바뀌었을 때 느끼는 낯설음 때문에 변경된 UI의 효율이 일시적으로 급락하는 경우이다.

두 번째로 표본 크기와 통계적 유의성이 있다.
단순히 "B안의 클릭률이 1% 더 높다"라고 해서 성공을 단정 지을 수 없다. 그 차이가 우연에 의한 것인지, 실제 효과인지 검증해야 한다. 사전에 검정력과 유의수준을 고려하여, 결과를 신뢰할 수 있는 최소 표본 크기를 계산하고 테스트를 시작해야 한다.

세 번째로 외부 요인 통제가 있다. 테스트 기간 동안 특정 이벤트가 발생하면 데이터가 오염될 수 있다. 대조군과 실험군은 반드시 동일한 기간에 실험이 진행되어야 하며, 주말과 평일의 사용자 패턴이 다르므로 최소 1주일 단위로 테스트 주기를 설정하는 것이 일반적이다.



---



 A/B TEST를 하기 위해서는 1-pager를 작성해야합니다.  

1-pager란 문제 정의, 관찰하고자하는 실험대상, 가설설정, 평가지표 등을 작성하는  
간단히 말해 계획서를 의미하는데요,

다른 부서와의 소통을 위한 역할 또,과거에 어떤 테스트를
어떻게 진행했는지에 대한 기록의 역할을 하기도 합니다.


 A/B TEST의 1-pager는 복잡한 실험 과정을 한눈에 파악하고, 의사결정권자가 "그래서 이 실험을 왜 했고, 결과가 무엇이며, 무엇을 할 것인가?"를 바로 이해할 수 있도록 작성하는 것이 핵심입니다.

예시를 통해 어떤 항목이 있고, 각 항목에 어떤 내용을 채우는 건지 보여드릴게요.



---



**예시**   
(푸는 거 아니고, 예시입니다! 읽으면서 이런 부분은 이런 내용을 채우는 거군... 이해하시면 돼요!)  




 📊 [ThinkGood] 메인 홈 검색 모듈 상위 노출 실험 1-Pager

---

**1. 실험 전략 및 배경 (Strategy & Observation)**

| 항목 | 내용 |
| :--- | :--- |
| **Target User** | 특정 공모전을 찾으려는 목적이 뚜렷하지만, 현재 하단에 위치한 검색바를 즉각적으로 발견하지 못한 유저 |
| **Observation** | **[데이터 분석 결과]** <br> • 검색 이용 유저의 상세 진입 전환율(CVR)은 40%, 미이용 유저의 전환율은 25%임 <br> • 현재 검색 이용률은 30%로, 검색 가시성 개선을 통한 전환율 상승 여력이 큼 |
| **Problem Statement** | 검색바가 배너 하단에 위치하여 시인성이 낮으며, 유저가 정보를 탐색하기 전 이탈할 확률이 높음 |
| **Hypothesis** | 하단 검색바를 페이지 **최상단으로 노출**시켜 검색 접근성을 높인다면, 전체 공모전 상세 진입률(CVR)이 개선될 것이다 |

---

 **2. 실험군 설계 (Experimental Groups)**

<table>
  <tr>
    <td align="center" width="50%">
      <b>A (Control) - 기존안</b><br>
      <img src="https://i.postimg.cc/kgcLc7Qy/Gemini-Generated-Image-enqvuhenqvuhenqv-(1).png" width="100%"><br>
      <sub>검색바 하단 위치 (기존 UI)</sub>
    </td>
    <td align="center" width="50%">
      <b>B (Variant) - 실험안</b><br>
      <img src="https://i.postimg.cc/0N51Rz5M/Gemini-Generated-Image-enqvuhenqvuhenqv.png" width="100%"><br>
      <sub>검색바 최상단 배치 (변경 UI)</sub>
    </td>
  </tr>
</table>

---

 **3. 지표 및 통계 설계 (Metrics & Statistics)**

> **핵심 측정 지표**
> * **Primary Metric (주지표):** 공모전 상세 페이지 진입률 (Detail Page CVR)
> * **Secondary Metric (보조지표):** 검색바 클릭률 (Search CTR)
> * **Guardrail Metric (방어지표):** 메인 페이지 이탈률 (Bounce Rate)

| 통계적 설계 항목 | 설정값 | 비고 |
| :--- | :--- | :--- |
| **Significance Level ($\alpha$)** | 5% | 유의수준 0.05 |
| **Statistical Power ($1-\beta$)** | 80% | 검정력 유지 |
| **Minimum Sample Size** | **총 92,000명** | 각 그룹당 46,000명 할당 |
| **Required Period** | **최소 4일** |  

---

**4. 위험 관리 및 중단 기준 (Risk & Andon)**

* **Trade-Off:**   
    검색바 상단 노출로 인해 메인 배너의 시각적 주목도가 분산될 수 있으며, 기존 UI에 익숙한 유저에게 일시적인 초두 효과(Primacy Effect)에 의한 혼란이 발생할 수 있음.
* **Andon (중단 기준):**
    1. 대조군 대비 전체 상세 진입률이 **5% 이상 유의미하게 하락**할 경우
    2. UI 변경으로 인한 페이지 오류나 유저 이탈률이 **3일 연속 급증**할 경우

---







<br>

# Q2. 1-pager를 작성해봅시다. (꼬리문제 3 + 보고서 채우는 것1)
저희가 만든 데모 사이트 링크를 첨부해드릴테니, 마음껏 보신 뒤에 위의 예시처럼 1-pager를 작성해보기 바랍니다.

여기서 실험군 설계에 들어가는 사진은 굳이 넣을 필욘 없고 말로만 설명해주셔도 됩니다. 아니면 gemini에게 요청해볼 수도 있겠네요!  

또한 문제탐색에 필요한 cvr과 같은 지표들은 직접 가정하고 하시면 됩니다. 다만, 해당 지표들이 큰 문제가 있는 지표여야... a/b test를 진행하는 이유가 되겠죠!?


정리해보면, 저희 웹사이트 보시고 '이건 여기에 배치하면 더 좋을 거 같은데?' 하는 것들을 찾아서 모든 것을 가정한 뒤에 쓰시면 됩니다!

[데모 웹사이트](https://huggingface.co/spaces/HanJaeHui/ThinkGood_Recommendation_System)


추가적으로 아래 실험군 항목에 사진을 넣으실 분들은 제가 아래 첨부드리는 사이트에 방문하셔서 해당 이미지를 코랩에 넣을 수 있는 형태로 변환한 링크를 복붙하시면 됩니다.  

(.png로 끝나는 링크 복붙하시면 됩니다.)

[포스트 이미지 사이트](https://postimages.org/)



<br>

## Q2-1. 실험군을 설계 해봅시다.
아래 보고서를 채워 주세요.

## A2-1. 아래 보고서 항목 설명을 참고하여 '[ ]' 부분을 지우고 채워주세요.  
  



📊 [ThinkGood] 실험 기획안 1-Pager

---

1. 실험 전략 및 배경 (Strategy & Observation)

| 항목 | 내용 |
| :--- | :--- |
| **Target User** | 공모전 추천을 받고 싶어 접속했으나, 입력 항목이 많거나 복잡해 보여 입력을 주저하는 대학생 유저 |
| **Observation** | **데이터 근거** <br> • 페이지 방문자(PV) 대비 'Submit' 버튼 클릭률(CTR)이 15%로 낮음 <br> • 입력 도중 이탈하는 유저가 전체 방문자의 60%를 차지함 |
| **Problem Statement** | 좌측 입력 패널(User Input)의 항목들이 단순히 나열되어 있어, 유저가 입력에 대한 부담(Cognitive Load)을 느끼고 이탈함 |
| **Hypothesis** | **입력 폼 상단에 "3초 만에 내 맞춤 공모전 찾기"라는 명확한 가치(Value Proposition)를 제시하고, 'Submit' 버튼을 강조하면 입력 완료율이 상승할 것이다**  |

---

 2. 실험군 설계 (Experimental Groups)

<table>
  <tr>
    <td align="center" width="50%">
      <b>A (Control) - 기존안</b><br>
      <img src="[기존 시안 이미지 URL 입력]" width="100%"><br>
      <sub>기존 Hugging Face 기본 스타일.회색 배경의 사이드바에 'User Input'이라는 단순한 헤더와 기본 크기의 'Submit' 버튼 배치</sub>
    </td>
    <td align="center" width="50%">
      <b>B (Variant) - 실험안</b><br>
      <img src="[변경 시안 이미지 URL 입력]" width="100%"><br>
      <sub>사이드바 최상단에 "🏆 딱 맞는 공모전 찾아드려요!" 문구 추가. 'Submit' 버튼의 색상을 **밝은 파란색(Brand Color)**으로 변경하고 텍스트를 **"추천 결과 확인하기"**로 변경하여 행동을 유도</sub>
    </td>
  </tr>
</table>


<br>

## Q2-2. 실험에 필요한 최소 test users 수를 구해봅시다.
유의수준 0.05, 검정력 80% 하에 수행해주세요.  


아래 공식 참고해서, 최소 test users 수를 구하는 코드 작성해주세요.

추가적으로 n은 산출하는데에 가정한 p1, p2도 같이 작성해주세요.

$$n = \frac{(Z_{\alpha/2} + Z_{\beta})^2 \cdot [p_1(1-p_1) + p_2(1-p_2)]}{(p_1 - p_2)^2}$$

## A2-2. 아래 질문에 대한 답과 코드를 채워주세요.

p1, p2는 각각 무엇을 의미하는 확률이고, 어떤 값으로 가정하셨나요?   
또한 위 공식을 보고 n을 구하기 위한 코드를 작성해주세요. 단, 라이브러리는 z값을 구하는 용도의 것만 사용해주세요.


### 답변:

*   p1 (기존 전환율): 현재 데모 사이트의 메인 홈에서 공모전 상세 페이지로 진입하는 비율(CVR)을 의미한다. 기존 방문자 대비 Submit 비율인 15% (0.15)로 가정했다.
*   p2 (목표 전환율): 개선된 UI를 적용했을 때 기대하는 최소한의 전환율이다. UI 개선을 통해 17% (0.17)까지 2%p 상승시키는 것을 목표로 가정했다.

In [1]:
# A2-2 코드작성. 나온 n은 아래 Minimum Sample Size에 기재.

import math
from scipy.stats import norm

# 1. 설정값 입력
alpha = 0.05  # 유의수준 (Type I error)
beta = 0.2    # 1 - 검정력 (Type II error, Power=0.8)
p1 = 0.15     # 대조군(A) 예상 전환율 (15%)
p2 = 0.17     # 실험군(B) 예상 전환율 (17%)

# 2. Z값 구하기
z_alpha = norm.ppf(1 - alpha / 2) # 양측 검정
z_beta = norm.ppf(1 - beta)

# 3. 최소 표본 수(n) 계산 공식 적용
# n = (Z_alpha/2 + Z_beta)^2 * [p1(1-p1) + p2(1-p2)] / (p1 - p2)^2
numerator = (z_alpha + z_beta)**2 * (p1 * (1 - p1) + p2 * (1 - p2))
denominator = (p1 - p2)**2
n = numerator / denominator

print(f"그룹당 필요한 최소 표본 수 (n): {math.ceil(n)}명")
print(f"총 필요한 최소 표본 수 (Total): {math.ceil(n) * 2}명")

#########

그룹당 필요한 최소 표본 수 (n): 5271명
총 필요한 최소 표본 수 (Total): 10542명


<br>

## Q2-3. 실험에 필요한 최소 실험 기간을 구해봅시다.

이 실험에 필요한 최소 test users가 92000명이 나왔고  
해당 웹페이지 하루 평균 방문자수가 50000명 정도라고 할 때 우리는 최소 4일이 필요합니다.

비교하고 싶은 대상이 총 두개이고, 하나의 대상을 실험하기 위해선 최소 이틀은 필요하기 때문이죠!

위에서 구하신 test users수를 바탕으로, 하루 평균 방문자수를 가정한 뒤 기간을 설정해주세요.  
구하신 기간은 밑에 표에 기재해주시면 됩니다.



## A2-3. 아래 세질문에 답하신 뒤 아래 보고서를 채워주세요.

가정한 하루 평균 방문자 수? 1,000명

산출된 최소 test users 수? 10,542명
  
그래서 필요한 최소 기간은? 약 11일 (10,542 ÷ 1,000 = 10.54일)

## **아래 보고서도 채워주세요!**  
**'[ ]' 부분을 지우고 해당 내용에 부합하는 내용들을 채워주시면 됩니다.**

---

**3. 지표 및 통계 설계 (Metrics & Statistics)**

> **핵심 측정 지표**
> * **Primary Metric (주지표):** 공모전 결과 조회 전환율 (Submit CVR)
> * **Secondary Metric (보조지표):** 입력 폼 평균 체류 시간 (Time on Page)
> * **Guardrail Metric (방어지표):** 결과 페이지 로딩 속도 및 오류율

| 통계적 설계 항목 | 설정값 | 비고 |
| :--- | :--- | :--- |
| **Significance Level ($\alpha$)** | 5% | 보통 5%(0.05) 설정 |
| **Statistical Power ($1-\beta$)** | 80% | 보통 80% 설정 |
| **Minimum Sample Size** | **총 5271명** | 그룹당 인원수 명시 |
| **Required Period** | **최소 11일** | 실험 진행 예정 기간 |

---

 **4. 위험 관리 및 중단 기준 (Risk & Andon)**

* **Trade-Off :**    
    입력 유도를 위해 문구를 너무 자극적으로 변경할 경우, 유저가 대충 입력하여 추천 결과의 정확도(만족도)가 떨어질 수 있음

* **Andon (중단 기준) :**
    1. 주지표(결과 조회 전환율)가 대조군 대비 10% 이상 급락할 경우 즉시 중단
    2. 서버 오류(500 Error) 발생률이 전체 트래픽의 1%를 초과할 경우 롤백

---

<br>

# Q3. A/B TEST를 수행해봅시다.

1-pager를 작성했으면 직접 test도 수행해봐야겠죠!  
위에 제가 만들었던 1-pager를 바탕으로 검정을 해볼게요.  
대표적인 검정방법에는 t-test가 있고 카이제곱 겁정이 있습니다.  


---



[실험 가설]  
귀무가설 ($H_0$): 메인 홈 검색 모듈의 위치(하단 vs 최상단)와 상세 페이지 진입 여부 사이에는 상관관계가 없다. 즉, 두 그룹의 상세 페이지 진입률($CVR$)은 동일하다.$$H_0: p_A = p_B$$대립가설 ($H_1$): 메인 홈 검색 모듈을 최상단에 노출하는 것은 상세 페이지 진입 여부에 영향을 미친다. 즉, 실험군(B)의 상세 페이지 진입률이 대조군(A)보다 높을 것이다.$$H_1: p_A < p_B$$


<br>

## Q3-1. 위 가설에 적합한 검정 이름과 왜 해당 검정을 써야하는지 이유도 작성해주세요. (t test와 카이제곱검정 중)

## A3-1.  검정 이름과 이유를 설명해주세요.

답변: 위 가설에 필요한 검정은 카이제곱 검정 이고,  

이유는 비교하고자 하는 데이터가 '상세 페이지 진입 성공/실패'라는 범주형 변수이며, 두 집단 간의 '비율' 차이가 유의미한지 검증해야 하기 때문이다.

<br>

In [2]:
'''''''''''''''''''''''''''''''''''''''
제가 세운 A/B TEST 설계 실험 코드입니다.
'''''''''''''''''''''''''''''''''''''''

import pandas as pd
from scipy.stats import chi2_contingency

# 1. 관측 데이터 입력 (실제 실험 결과 수치로 변경하세요)
# [상세진입 성공 수, 상세진입 실패 수]
control_results = [11500, 34500]  # 대조군(A)
variant_results = [12190, 33810]  # 실험군(B)

# 2. 분할표(Contingency Table) 생성
data = [control_results, variant_results]

# 3. 카이제곱 검정 수행
chi2, p_value, dof, expected = chi2_contingency(data)

# 4. 결과 출력
print(f"카이제곱 통계량: {chi2:.4f}")
print(f"P-value: {p_value:.10f}")

alpha = 0.05
if p_value < alpha:
    print(f"결과: P-value가 {alpha}보다 작으므로 귀무가설을 기각합니다.")
    print("=> 검색바 최상단 노출이 상세 페이지 진입률에 유의미한 변화를 주었습니다.")
else:
    print(f"결과: P-value가 {alpha}보다 크므로 귀무가설을 기각할 수 없습니다.")
    print("=> 검색바 위치 변경의 효과가 통계적으로 증명되지 않았습니다.")

카이제곱 통계량: 26.9884
P-value: 0.0000002047
결과: P-value가 0.05보다 작으므로 귀무가설을 기각합니다.
=> 검색바 최상단 노출이 상세 페이지 진입률에 유의미한 변화를 주었습니다.


실제 데이터가 없어서, 저도 필요한 데이터는 가정했답니다.  

이제 25기 분들의 A/B TEST를 수행해보시죠!

<br>

## Q3-2. Q2에서 작성한 실험 수행을 위해 가설을 설정해주세요.


## A3-2. 아래 두 가설 내용을 채워주세요.
  
[실험 가설]  
귀무가설 ($H_0$): 입력 폼의 문구와 버튼 디자인 변경은 결과 조회 전환율(Submit Rate)에 영향을 주지 않는다.
즉, 대조군(A)과 실험군(B)의 전환율은 통계적으로 동일하다. ($p_A = p_B$)
  
대립가설 ($H_1$): 입력 폼의 문구와 버튼 디자인 변경은 결과 조회 전환율을 상승시킨다.
즉, 실험군(B)의 전환율이 대조군(A)보다 높을 것이다. ($p_A < p_B$)

<br>

## Q3-3. 위 가설을 수행하기 위해 어떤 검정 방법을 사용해야하고 이유는 무엇인지 작성해주세요.

## A3-3. 아래 검정 이름과 이유를 채워주세요.
답변: 위 가설에 필요한 검정은 카이제곱 검정이고,  

이유는 우리가 비교하고자 하는 데이터가 '전환(성공) vs 미전환(실패)'이라는 범주형 데이터의 비율 차이이기 때문이다.

<br>

## Q3-4. 위에서 답한 것을 토대로, 검정을 수행해주세요. (t test와 카이제곱 검정 중)


## A3-4 아래 셀에 코드 작성해주세요. 그리고!!!
그리고 나온 값을 통해 수행하신 a/b test 결과를 설정하신 가설에 맞게 해석해주세요.

In [3]:
### A3-4. 코드작성

import pandas as pd
from scipy.stats import chi2_contingency

# 1. 관측 데이터 입력 (Q2에서 설계한 표본 수와 전환율을 바탕으로 가정)
# [전환 성공(Submit), 전환 실패]
# 대조군(A): 약 5,271명 중 15% 전환 가정 -> 약 791명 성공
control_results = [791, 4480]

# 실험군(B): 약 5,271명 중 17.5% 전환 가정 (목표치인 17%를 상회했다고 가정) -> 약 922명 성공
variant_results = [922, 4349]

# 2. 분할표(Contingency Table) 생성
data = [control_results, variant_results]

# 3. 카이제곱 검정 수행
chi2, p_value, dof, expected = chi2_contingency(data)

# 4. 결과 출력
print(f"카이제곱 통계량: {chi2:.4f}")
print(f"P-value: {p_value:.10f}")

alpha = 0.05
if p_value < alpha:
    print(f"결과: P-value가 {alpha}보다 작으므로 귀무가설을 기각합니다.")
    print("=> 입력 폼 문구 및 버튼 디자인 변경이 전환율에 유의미한 상승을 이끌어냈습니다.")
else:
    print(f"결과: P-value가 {alpha}보다 크므로 귀무가설을 기각할 수 없습니다.")
    print("=> 디자인 변경의 효과가 통계적으로 유의미하지 않습니다.")

#############

카이제곱 통계량: 11.7799
P-value: 0.0005987437
결과: P-value가 0.05보다 작으므로 귀무가설을 기각합니다.
=> 입력 폼 문구 및 버튼 디자인 변경이 전환율에 유의미한 상승을 이끌어냈습니다.


**결과해석**

답변:

1. 통계적 유의성 확인본 실험의 카이제곱 검정 결과, P-value가 0.000598로 유의수준 0.05보다 훨씬 작게 나타났다. 이는 대조군(A)과 실험군(B)의 전환율 차이가 단순히 우연에 의해 발생했을 확률이 0.06% 미만임을 의미하며, 따라서 통계적으로 매우 유의미한 차이가 있음이 증명되었다.

2. 가설 검증 결

*   귀무가설($H_0$): 기각
*   대립가설($H_1$): 채택

*   결론: 입력 폼 상단의 가치 제안 문구("3초 만에 내 맞춤 공모전 찾기")와 'Submit' 버튼의 색상 및 텍스트 변경은 유저의 전환율(Submit Rate)을 높이는 데 실제로 효과가 있다.


3. 비즈니스 인사이트 및 제언

*   심리적 허들 감소: 단순한 정보 나열보다 유저가 얻을 수 있는 '즉각적인 보상(3초 만에 찾기)'을 명시한 것이 유저의 인지적 부하(Cognitive Load)를 줄이고 참여를 유도했다.

*   시각적 강조의 효과: 무채색 계열의 기본 버튼보다 브랜드 컬러를 활용한 강조가 유저의 시선을 끌어 최종 행동(Click)으로 연결되는 데 기여했다.

*   향후 계획: 실험군(B)의 UI를 정식 업데이트로 반영할 것을 권장하며, 이후 단계에서는 입력 항목 자체를 줄여 전환율을 추가로 개선할 수 있는 후속 테스트를 제안한다.

<br>

**1주차 기초통계 과제는 여기까지였습니다. 수고하셨습니다!**